### DISCUSSION TITLE
For 1-N problems, how to customize the json returned structure through `Custom Select`

### DISCUSSION THREAD

**Discussion by bingryan** [SeaQL/sea-orm#1215] [Q&A]
language=MySQL

## Description

```sql
// table model 
pub struct Article {
    pub id: u64,
    pub user_id: u64,
    pub title: Option<String>,
    pub description: Option<String>,
    pub body: Option<String>,
}


pub struct User {
    pub id: u64,
    pub username: Option<String>,
    pub bio: Option<String>,
    pub image: Option<String>,
}


// custom select result 
pub struct ArticleQuery {
    pub title: Option<String>,
    pub description: Option<String>,
    pub body: Option<String>,
    pub author: Profile,
}


pub struct Profile {
    pub username: Option<String>,
    pub bio: Option<String>,
    pub image: Option<String>,
}

```

When I visit [1-N wiki](https://www.sea-ql.org/SeaORM/docs/next/relation/one-to-many/), 

I only see the relationship between the definition table and the table, but I didn't see 1-N structure is returned(eg. ArticleQuery).

**Comment by billy1624** [comment]
Hey @bingryan, sorry for the delay. At this stage, you need to select Article and Profile individually and then combine it afterward.

The difficulties actually comes from the Rust's type system: https://www.sea-ql.org/sea-orm-cookbook/001-eager-load-nested-relation.html

This issues, https://github.com/SeaQL/sea-orm/issues/1191, will address this in the near future.

Btw... do checkout [seaography](https://github.com/SeaQL/seaography) where it provide you GraphQL layer on top of SeaORM with 1+N loading in batch out of the box: https://www.sea-ql.org/blog/2022-09-27-getting-started-with-seaography/#query-data-via-graphql

**Reply by bingryan** [reply]
Thanks @billy1624
